# covid_meta
Software and codes used for COVID gut microbiome meta-analysis

Bioinformatic anlysis 'bioinformatics.sh' in bioinformatics_statistics

TrimGalore v0.6.7, HUMAnN 3, MetaPhlAn 3 and Kraken2 can be installed in conda; SPINGO v1.3 in Linux system.

QC with TrimGalore v0.6.7 "1. System requirements": Cutadapt and FastQC; "2. Installation guide": https://github.com/FelixKrueger/TrimGalore; "3. Demo and 4. Instructions for use": https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md

16S taxonomy using SPINGO v1.3 "System requirements, Installation guide, Demo, Instructions for use": https://github.com/GuyAllard/SPINGO 

Shotgun function using HUMAnN 3 "System requirements, Installation guide, Demo, Instructions for use": https://github.com/biobakery/humann 

Shotgun taxonomy using MetaPhlAn 3 "System requirements, Installation guide, Demo, Instructions for use": https://github.com/biobakery/MetaPhlAn 

Shotgun taxonomy using Kraken2 "System requirements, Installation guide, Demo, Instructions for use": https://github.com/DerrickWood/kraken2/wiki/Manual


R codes in bioinformatics_statistics (NMDS.R, PERMANOVA.R, dysbiosis.R, association_heatmap.R, RandomForest_COVIDSeverityIndex.R) used in the analysis. 
R v3.6.2 "System requirements, Installation guide, Demo, Instructions for use": https://cran.r-project.org/manuals.html 
